op-node/rollup/derive/frame_queue_test.go

```diff
@@ -0,0 +1,154 @@
+package derive
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"log/slog"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive/mocks"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPruneFrameQueue(t *testing.T) {
+	for _, tt := range []struct {
+		desc string
+		fs   []testFrame
+		exp  []testFrame
```

Comment:
What is `fs` and `exp`? Oh, frames and expected. Man I hate go's penchant for impossible to understand names. &lt;/old-man-shouts-at-cloud>

Reply:
I'm guilty - I love Go's penchant for the shortest possible names. And see, you guessed correctly! I'll expand the names nonetheless :)